test/parallel/test-whatwg-encoding-textencoder.js

```diff
@@ -2,50 +2,93 @@
 'use strict';
 
 const common = require('../common');
-
 const assert = require('assert');
 const { TextDecoder, TextEncoder } = require('util');
 const { customInspectSymbol: inspect } = require('internal/util');
 
-const encoded = Buffer.from([0xef, 0xbb, 0xbf, 0x74, 0x65,
-                             0x73, 0x74, 0xe2, 0x82, 0xac]);
+// This test checks TextEncoder's instantiation, default encoding,
+// encoding capabilities, and conformity with util.inspect().
 
-// Make Sure TextEncoder exists
-assert(TextEncoder);
+{
+  const textEncoder = new TextEncoder();
+  assert.ok(textEncoder);
+  assert.strictEqual(textEncoder instanceof TextEncoder, true);
+}
 
-// Test TextEncoder
 {
-  const enc = new TextEncoder();
-  assert.strictEqual(enc.encoding, 'utf-8');
-  assert(enc);
-  const buf = enc.encode('\ufefftest€');
-  assert.strictEqual(Buffer.compare(buf, encoded), 0);
+  const textEncoder = new TextEncoder();
+  assert.strictEqual(textEncoder.encoding, 'utf-8');
+}
+
+function checkIfBuffersAreEqual(buffer1, buffer2) {
+  const comparisonResult = Buffer.compare(buffer1, buffer2);
+  return comparisonResult === 0;
 }
 
 {
-  const enc = new TextEncoder();
-  const buf = enc.encode();
-  assert.strictEqual(buf.length, 0);
+  const EURO_CURRENCY_SIGN = '\u20AC';
+  const testString = 'test' + EURO_CURRENCY_SIGN;
+
+  const textEncoder = new TextEncoder();
+  const bufferForEncodedTestString = textEncoder.encode(testString);
+  const expectedBuffer = Buffer.from(testString, 'utf8');
+
+  const encodingWasSuccessful = checkIfBuffersAreEqual(
+    bufferForEncodedTestString,
+    expectedBuffer
+  );
+
+  assert.strictEqual(encodingWasSuccessful, true);
 }
 
 {
-  const enc = new TextEncoder();
-  const buf = enc.encode(undefined);
-  assert.strictEqual(buf.length, 0);
+  const textEncoder = new TextEncoder();
+  const bufferForEmptyEncoding = textEncoder.encode();
+  assert.strictEqual(bufferForEmptyEncoding.length, 0);
 }
 
 {
-  const fn = TextEncoder.prototype[inspect];
-  assert.doesNotThrow(() => {
-    fn.call(new TextEncoder(), Infinity, {});
-  });
-
-  [{}, [], true, 1, '', new TextDecoder()].forEach((i) => {
-    common.expectsError(() => fn.call(i, Infinity, {}),
-                        {
-                          code: 'ERR_INVALID_THIS',
-                          type: TypeError,
-                          message: 'Value of "this" must be of type TextEncoder'
-                        });
-  });
+  const textEncoder = new TextEncoder();
+  const bufferForUndefinedEncoding = textEncoder.encode(undefined);
+  assert.strictEqual(bufferForUndefinedEncoding.length, 0);
 }
+
+const textEncoderInspectFunc = TextEncoder.prototype[inspect];
+const recursiveInspectDepth = Infinity;
+const inspectOptions = {};
+
+assert.doesNotThrow(() => {
+  textEncoderInspectFunc.call(
+    new TextEncoder(),
+    recursiveInspectDepth,
+    inspectOptions
```

Comment:
Using variables for `textEncoderInspectFunc`, `Infinity` and empty object isn't particularly helpful and in fact reduces legibility of this test.

Reply:
How does variable abstraction reduce legibility? Unless the variable abstraction add many layers or the variable name is inaccurate, I disagree.

> And I realize this also applies to the old fn, which was IMO an unnecessary variable.

Where is the line between "unnecessary" and "helpful"? Several of the test's original variables are unnecessary. Literal values could replace them. Yet, I think their inclusion was helpful.

The old `fn` was unhelpful to me. That's only because the name is vague.